costs/include/quadratic_cost.h

```diff
@@ -0,0 +1,85 @@
+#ifndef TORC_QUADRATIC_COST_H
+#define TORC_QUADRATIC_COST_H
+
+#include <string>
+#include "base_cost.h"
+
+namespace torc {
+    /**
+     * Class implementation of a linear cost function, f(x) = x^T A x, where A is a symmetric matrix.
+     * @tparam scalar_t the type of scalar used for the cost
+     */
+    template <class scalar_t>
+    class QuadraticCost: public BaseCost<scalar_t> {
+        using vectorx_t = Eigen::VectorX<scalar_t>;
+        using matrixx_t = Eigen::MatrixX<scalar_t>;
+
+    public:
+        /**
+         * Constructor for the quadratic cost class
+         * @param coefficients the A in x^T A x.
```

Comment:
let's actually make the cost of the form 0.5*x^T^A*x (so adding the 0.5 out front) as that is normal in optimization. Then we need to adjust the below calculations.